### ISSUE TITLE
Fail to generate universal apks from aab on gradle 5

### ISSUE THREAD

**Issue by fadeltd** [google/bundletool#113]
**Describe the bug**
I tried to build generate universal apks from aab file after upgrading gradle version 5.4.1, I can still generate apks mode universal when I use gradle version 4.10.3

**Bundletool version(s) affected**
Version: 0.10.2

**Stacktrace**
```
Error: Program type already present: $r8$java8methods$utility$Integer$hashCode$II
[BT:0.10.2] Error: Dex merging failed.
com.android.tools.build.bundletool.model.exceptions.CommandExecutionException: Dex merging failed.
	at com.android.tools.build.bundletool.mergers.D8DexMerger.translateD8Exception(D8DexMerger.java:98)
	at com.android.tools.build.bundletool.mergers.D8DexMerger.merge(D8DexMerger.java:77)
	at com.android.tools.build.bundletool.mergers.ModuleSplitsToShardMerger.mergeDexFiles(ModuleSplitsToShardMerger.java:334)
	at com.android.tools.build.bundletool.mergers.ModuleSplitsToShardMerger.lambda$mergeDexFilesAndCache$2(ModuleSplitsToShardMerger.java:302)
	at java.util.HashMap.computeIfAbsent(HashMap.java:1127)
	at com.android.tools.build.bundletool.mergers.ModuleSplitsToShardMerger.mergeDexFilesAndCache(ModuleSplitsToShardMerger.java:300)
	at com.android.tools.build.bundletool.mergers.ModuleSplitsToShardMerger.mergeSingleShard(ModuleSplitsToShardMerger.java:194)
	at com.android.tools.build.bundletool.mergers.ModuleSplitsToShardMerger.merge(ModuleSplitsToShardMerger.java:104)
	at com.android.tools.build.bundletool.splitters.BundleSharder.shardBundle(BundleSharder.java:103)
	at com.android.tools.build.bundletool.splitters.ShardedApksGenerator.generateSplits(ShardedApksGenerator.java:70)
	at com.android.tools.build.bundletool.commands.BuildApksManager.executeWithZip(BuildApksManager.java:179)
	at com.android.tools.build.bundletool.commands.BuildApksManager.execute(BuildApksManager.java:110)
	at com.android.tools.build.bundletool.commands.BuildApksCommand.execute(BuildApksCommand.java:524)
	at com.android.tools.build.bundletool.BundleToolMain.main(BundleToolMain.java:74)
	at com.android.tools.build.bundletool.BundleToolMain.main(BundleToolMain.java:46)
Caused by: com.android.tools.r8.CompilationFailedException: Compilation failed to complete
	at com.android.tools.r8.utils.ExceptionUtils.withCompilationHandler(ExceptionUtils.java:76)
	at com.android.tools.r8.utils.ExceptionUtils.withD8CompilationHandler(ExceptionUtils.java:45)
	at com.android.tools.r8.D8.run(D8.java:67)
	at com.android.tools.build.bundletool.mergers.D8DexMerger.merge(D8DexMerger.java:70)
	... 13 more
Caused by: com.android.tools.r8.utils.AbortException
	at com.android.tools.r8.utils.Reporter.failIfPendingErrors(Reporter.java:77)
	at com.android.tools.r8.utils.Reporter.fatalError(Reporter.java:58)
	at com.android.tools.r8.utils.ExceptionUtils.withCompilationHandler(ExceptionUtils.java:67)
	... 16 more
```

**To Reproduce**
`./bundletool build-apks --bundle=app/build/outputs/bundle/debug/app.aab --output=app/build/outputs/bundle/debug/app.apks --mode=universal`

**Expected behavior**
Universal APKS generated from aab file

**Environment:**
OS: macOS 10.14.5

**Additional context**
I found the same error [android studio blog](https://androidstudio.googleblog.com/2019/02/android-studio-34-beta-4-available.html) and [goole issue tracker](https://issuetracker.google.com/issues/123242448) but different stacktrace

**Comment by kvafy** [comment]
Could you please try updating the r8 version in build.gradle (`"com.android.tools:r8:1.0.37"`, latest is `1.5.68`), build bundletool from source and check whether the issue is still present when running build-apks?

**Comment by firoze** [comment]
> maybe you can try the executableJar from my PR [bundletool-all.jar.zip](https://github.com/google/bundletool/files/3608281/bundletool-all.jar.zip)
> 
> Currently, updating the r8 in bundletool is the only workaround I found

Thank you very much! I was able to build a universal `.apks` file from your PR's executableJar. 
I've at least been temporarily unblocked for now. Hoping your PR is merged soon and a release is out.

**Comment by plecesne** [comment]
We've updated the version of R8 in 0.10.3. Could you give it a try?

**Comment by fadeltd** [comment]
> We've updated the version of R8 in 0.10.3. Could you give it a try?

it works

**Comment by rallat** [comment]
We had the same issue #113 with bundletool 0.10.2 but the 3 days ago released [0.10.3](https://github.com/google/bundletool/releases/tag/0.10.3) fixed it for us. Thanks!
